dotnet/ios/ios12/ARKitVision/ARKitVision/StatusViewController.cs

```diff
@@ -0,0 +1,146 @@
+
+/*
+ Displayed at the top of the main interface of the app that allows users to see
+ the status of the AR experience, as well as the ability to control restarting
+ the experience altogether.
+ */
+
+using ARKit;
+
+namespace ARKitVision;
+
+/// <summary>
+/// Utility class for showing messages above the AR view.
+/// </summary>
+public partial class StatusViewController : UIViewController
+{
+        /// Seconds before the timer message should fade out. Adjust if the app needs longer transient messages.
+        private const double DisplayDuration = 6d;
+
+        private Dictionary<MessageType, NSTimer> timers = new Dictionary<MessageType, NSTimer> ();
+
+        // Timer for hiding messages.
+        private NSTimer? messageHideTimer;
+
+        public StatusViewController (IntPtr handle) : base (handle) { }
```

Comment:
This doesn't have to be public:

```suggestion
        protected StatusViewController (IntPtr handle) : base (handle) { }
```